src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_dsa.c

```diff
@@ -156,11 +156,16 @@ int32_t CryptoNative_GetDsaParameters(
         assert(false);
 
         // since these parameters are 'out' parameters in managed code, ensure they are initialized
-        if (p) *p = NULL; if (pLength) *pLength = 0;
-        if (q) *q = NULL; if (qLength) *qLength = 0;
-        if (g) *g = NULL; if (gLength) *gLength = 0;
-        if (y) *y = NULL; if (yLength) *yLength = 0;
-        if (x) *x = NULL; if (xLength) *xLength = 0;
+        if (p) *p = NULL;
+        if (pLength) *pLength = 0;
+        if (q) *q = NULL;
+        if (qLength) *qLength = 0;
+        if (g) *g = NULL;
+        if (gLength) *gLength = 0;
+        if (y) *y = NULL;
+        if (yLength) *yLength = 0;
+        if (x) *x = NULL;
+        if (xLength) *xLength = 0;
```

Comment:
@bartonjs, I split these conditions, as GCC 9 on Clear Linux was saying:

```sh
/runtime/src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_dsa.c:163:9: error: this ‘if’ clause does not guard... [-Werror=misleading-indentation]
  163 |         if (x) *x = NULL; if (xLength) *xLength = 0;
      |         ^~
```

Hope this is ok. 🙂

Reply:
@bartonjs Why do we have this at all? There is `assert(false)` and this is unrechable code. Should this be just replaced by asserts that none of the arguments is null?

Reply:
I forget.. do these asserts exist in release builds?

I think this is here because we want to make sure that P/Invokes we declare with `out` maintain C# semantics (outs don't get skipped on bad data returns).  If the release build will terminate the process from an assert then I agree it's fully dead code and we can delete it.

If it's just a debug diagnostic, I feel a lot more comfortable with it being here.

Reply:
`out` C# semantics guarantees that nulls are not passed in. Unless you use unsafe code ... but then you can do anything.

And if nulls were passed into this method somehow, we would get a hard AV that terminates the process when trying to assign the argument.

I think these should be just change to `assert(y != NULL);`, etc.

Reply:
Should we change these to `assert` as part of this PR?

Reply:
I think it would be a good idea since you are touching this.

Reply:
Sure, I meant should I go ahead or wait for @bartonjs's feedback.

Reply:
@jkotas, just to make sure, is this a correct change:

```diff
     if (!dsa || !p || !q || !g || !y || !x)
     {
         assert(false);
+        assert(p != NULL);
+        assert(q != NULL);
+        assert(g != NULL);
+        assert(y != NULL);
+        assert(x != NULL);
 
         // since these parameters are 'out' parameters in managed code, ensure they are initialized
-        if (p) *p = NULL;
         if (pLength) *pLength = 0;
-        if (q) *q = NULL;
         if (qLength) *qLength = 0;
-        if (g) *g = NULL;
         if (gLength) *gLength = 0;
-        if (y) *y = NULL;
         if (yLength) *yLength = 0;
-        if (x) *x = NULL;
         if (xLength) *xLength = 0;
         return 0;
     }
```

Reply:
I'll concede to changing this validation block to asserts.  Especially since it looks like none of the "length" parameters got validated as non-null before they got written to in the main path.

Reply:
@am11 pretty sure it's replace the entirety of the `if (!dsa ...) { ... }` like

```C
int32_t CryptoNative_GetDsaParameters(
    const DSA* dsa,
    const BIGNUM** p, int32_t* pLength,
    const BIGNUM** q, int32_t* qLength,
    const BIGNUM** g, int32_t* gLength,
    const BIGNUM** y, int32_t* yLength,
    const BIGNUM** x, int32_t* xLength)
{
    assert(dsa != NULL);
    assert(p != NULL);
    assert(pLength != NULL);
    assert(q != NULL);
    assert(qLength != NULL);
    ...
    assert(x != NULL);
    assert(xLength != NULL);

    DSA_get0_pqg(dsa, p, q, g);
    ...
```

Reply:
Thanks. I have converted conditions to `assert` in three places which PR was touching.